### ISSUE TITLE
Exposing SIREN's theta_initial and theta parameters.

### ISSUE THREAD

**Issue by afiaka87** [lucidrains/deep-daze#36]
Update: The initial image generated for delta 1,2,3,4,5,...41
https://user-images.githubusercontent.com/3994972/106395379-9f77c380-63c7-11eb-96ba-ccda3da22334.mp4

As dicussed in another issue, deep-daze is sometimes overly sensitive to its initial input. In searching for a lead on how to resolve this issue, I decided to expose the SIREN hyperparameter denoted theta (I think) in the original notebook. So far I've found some interesting results, but I'm still working out what of it is useful. It's probably just best if I show you some examples of different theta on the same text input. For reference, I'm using @dginev seed and text, as he mentioned not being able to get a "cheery" output out of the rather cheery phrase "seed of hope". You can find that discussion here: https://github.com/lucidrains/deep-daze/issues/9#issuecomment-763249044

`--seed=872073`
`text: "seed of hope"`
`iterations: 1000`
`epochs: 1`
`image_width: 256`
`num_layers: 32`
`batch_size: 64`
`gradient_accumulate_every: 1`

This is the code I'm using to run through different theta parameters for the same input. Careful, this code will run 180 full 1000 iteration runs. 
```bash
#!/bin/bash

for ((theta=1;theta<180;theta++)); do
  echo "processing theta: " + $theta;
  imagine "seed of hope"\
     --seed=872072 \
     --num_layers=32 \
     --image_width=256 --save_progress=True --save_every=10 --epochs=1 \
     --batch_size=32 --gradient_accumulate_every=1 \
     --iterations=1000
     --overwrite=True \
     --theta_initial=$theta \
     --theta_hidden=$theta \
     --save_date_time=True;
  wait; # Important, otherwise the loop will continue before finishing and you'll run out of memory.
done

```
Here's the code if you'd like to try yourself. @lucidrains Would love your input on this. Not sure it's ready to be merged quite yet but let me know if you're interested in that. https://github.com/afiaka87/deep-daze/tree/theta_output_dir_params

## Early results:
This is what I was able to run on my RTX 2070 this morning. Each image represents a change in the theta value by 1. The final image is theta=56. As you can see, it definitely gives different results. But I'm still not sure what to make of it. 

![1_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393641-ef518d00-63bd-\11eb-95e9-f9fa4e150375.png)
![2_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393642-f24c7d80-63bd-11eb-9a69-502b74da12ed.png)
![3_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393645-faa4b880-63bd-11eb-894f-c4a8b7598b24.png)
![4_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393647-fc6e7c00-63bd-11eb-8e27-68f257c72843.png)
![5_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393650-fe383f80-63bd-11eb-8d74-3386513175c7.png)
![6_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393718-61c26d00-63be-11eb-9814-5166ffc69c9c.png)
![7_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393724-6c7d0200-63be-11eb-8533-9a5906b6d82c.png)
![8_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393727-6e46c580-63be-11eb-81e5-05716e92ada2.png)
![9_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393728-70a91f80-63be-11eb-8976-82a161db64f7.png)
![10_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393729-7272e300-63be-11eb-969f-693b5cf62022.png)
![11_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393730-743ca680-63be-11eb-90ef-7df636fc015f.png)
![12_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393735-76066a00-63be-11eb-993d-3ea13e20825a.png)
![13_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393741-7999f100-63be-11eb-8178-f54146824d96.png)
![14_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393742-7b63b480-63be-11eb-88b2-a6018ef8ddde.png)
![15_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393746-7e5ea500-63be-11eb-9c2e-1cc3fa950067.png)
![16_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393748-80c0ff00-63be-11eb-9e4b-5e162f27b3c2.png)
![17_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393750-83235900-63be-11eb-8d84-a3233f8d6bb2.png)
![18_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393756-8585b300-63be-11eb-8c13-ad84cab496e2.png)
![19_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393758-87e80d00-63be-11eb-930f-66e4cc001a4a.png)
![20_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393763-8b7b9400-63be-11eb-9942-b6935c542f97.png)
![21_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393764-8d455780-63be-11eb-8f6b-cfe10793df01.png)
![22_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393767-8fa7b180-63be-11eb-8a68-d1b5fae3e025.png)
![23_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393772-96362900-63be-11eb-8c52-9f91855b8fd9.png)
![24_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393774-9b937380-63be-11eb-9f45-a328cd180aa1.png)
![25_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393779-9fbf9100-63be-11eb-8d2e-89f87aa6486f.png)
![26_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393782-a221eb00-63be-11eb-91ee-0484b8e1a79d.png)
![27_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393784-a51cdb80-63be-11eb-92aa-8baa1af8f6d1.png)
![28_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393786-a51cdb80-63be-11eb-87dc-ad3558458012.png)
![29_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393787-a5b57200-63be-11eb-8e39-c3320f3dc19a.png)
![30_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393792-ac43e980-63be-11eb-9f6a-b266983fcb18.png)
![31_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393793-ac43e980-63be-11eb-82d0-9ce764e58cc3.png)
![32_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393794-acdc8000-63be-11eb-89c5-b8a689a9bd02.png)
![33_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393795-acdc8000-63be-11eb-823a-eee3def84812.png)
![34_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393796-acdc8000-63be-11eb-962d-4d035e70c114.png)
![35_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393797-acdc8000-63be-11eb-8e58-ebac80d255ea.png)
![36_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393798-ad751680-63be-11eb-8e6f-12b7d66168fb.png)
![37_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393799-ad751680-63be-11eb-922c-503bbc1dbc98.png)
![38_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393800-ad751680-63be-11eb-9b2e-d8cb8c9de4be.png)
![39_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393801-ad751680-63be-11eb-8a3e-99157b49b449.png)
![40_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393802-ae0dad00-63be-11eb-94a8-36c3c6a5f351.png)
![41_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393803-ae0dad00-63be-11eb-8a68-0fe6f9354731.png)
![42_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393804-ae0dad00-63be-11eb-87fe-6560681ba004.png)
![43_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393805-ae0dad00-63be-11eb-90db-3fa88ed9d7d0.png)
![44_seed_of_hope](https://user-images.githubusercontent.com/3994972/106393806-aea64380-63be-11eb-9e91-4e472f87edce.png)
![45_seed_of_hope](https://user-images.githubusercontent.com/3994972/106394598-0ba3f880-63c3-11eb-85c9-4357d40db9c3.png)
![47_seed_of_hope](https://user-images.githubusercontent.com/3994972/106394599-0c3c8f00-63c3-11eb-907a-1224e125ad50.png)
![48_seed_of_hope](https://user-images.githubusercontent.com/3994972/106394600-0c3c8f00-63c3-11eb-8cb4-b66baee68c53.png)
![49_seed_of_hope](https://user-images.githubusercontent.com/3994972/106394601-0c3c8f00-63c3-11eb-8673-91cb637f32ba.png)
![51_seed_of_hope](https://user-images.githubusercontent.com/3994972/106394603-0c3c8f00-63c3-11eb-8f0d-e05b6b42092b.png)
![54_seed_of_hope](https://user-images.githubusercontent.com/3994972/106394605-0cd52580-63c3-11eb-9899-cd75d93f6f4f.png)
![55_seed_of_hope](https://user-images.githubusercontent.com/3994972/106394606-0cd52580-63c3-11eb-8d8c-e42487f2f669.png)
![56_seed_of_hope](https://user-images.githubusercontent.com/3994972/106394607-0cd52580-63c3-11eb-9043-a73106ef801b.png)

**Comment by dginev** [comment]
It may end up useful...  I can now use the new setting allowing me to start off from a pre-existing image, so can see the generation starting off with very bright and "cheery" imagery when seeded accordingly. But, somewhat mysteriously, as the generation progresses there is a noticeable slide back into darker hues, even for super-positive keywords (sunny, summer, angel, etc). The technical causes are beyond me, but one wonders if this parameter may not be a part of the answer?

**Comment by lucidrains** [comment]
@afiaka87 @dginev released in 0.4.2! https://github.com/lucidrains/deep-daze/commit/0e879593cdecada727a40ddf7a4f55b58651f35c

**Comment by afiaka87** [comment]
Much obliged.
